DashAI/back/core/schema_fields/string_field.py

```diff
@@ -0,0 +1,65 @@
+from typing import List
+
+from pydantic import AfterValidator, Field
+from typing_extensions import Annotated
+
+
+def __check_choices(enum: List[str]):
+    """Factory to create custom validator for string field.
+    Checks if the input str is in the enum.
+
+    Parameters
+    ----------
+    enum: List[str]
+        All the posible string values of the param.
+
+    Returns
+    -------
+    str -> str
+        A function that inspect the input string.
+    """
+
+    def check_str_in_enum(x: str) -> str:
+        assert x in enum, f"{x} is not in the enum"
```

Comment:
Asserts should be used only for testing, not at runtime. This is because, unless you are running a testing session, they are disabled!

https://stackoverflow.com/questions/40182944/whats-the-difference-between-raise-try-and-assert

```suggestion
        if x not in enum:
            raise ValueError(f"{x} is not in the enum")
```

Reply:
Ok, i used assert beacause is used in the pydantic examples for validators, but if it can be removed is better to use Exceptions.